### PR TITLE
Track New Order Creation

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes 504-gh-Automattic/woocommerce-payments-server

#### Changes proposed in this Pull Request

* Adds the `ActionScheduler` library to `composer.json` (please note, this library is already bundled into WooCommerce, so it should already be installed on any site which uses WC Payments, however added here to provide code completion etc. as we are using it to send jobs to the server.
* Adds logic to create an ActionScheduler Job when an order is created via WooCommerce Payments.
* Adds logic to send an API request to the server via ActionScheduler.

#### Testing instructions

* Checkout the branch.
* Create an order in your store.
* Verify that you can see an action of type  `wcpay_track_new_order` in the **Tools -> Scheduled Actions** section of `wp-admin`. (It should be under `pending` actions if it hasn't already been run by AS).
* Click `Run` to run the action. Check the logs for your site in **WooCommerce -> Status -> Logs**. Verify that a POST request was made to the Payments API to the `/tracking/new_order` endpoint. Note that you'll also need the corresponding server PR to verify that the serverside logic is working as expected (but this PR only covers the client side, so just sending the request is the expected behaviour here).
* Confirm the unit tests pass with `npm run:test`

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payments-action-scheduler-service.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	 * This function is a hook that will be called by ActionScheduler when an order is created.
+	 * It will make a request to the Payments API to track this event.
+	 *
+	 * @param int   $order_id    The ID of the order that has been created.
+	 * @param array $order_data  The data for the order which has been created.
+	 *
+	 * @return bool
+	 */
+	public function track_new_order_action( $order_id, $order_data ) {
+		// @todo Do we need to add logging here of whether the API call was successful?
```
Comment (dmallory42) [inline includes/class-wc-payments-action-scheduler-service.php:57]: Would love to get thoughts on this. Is it necessary to log that the ActionScheduler job to send the data to the Payments API got called? If so, do we also want to know if it succeeded?

Comment (dmallory42) [reply includes/class-wc-payments-action-scheduler-service.php:57]: Discussed with @luizreis and this is already logged by the API client, so not needed here.

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
       "automattic/jetpack-connection": "1.20.0",
       "automattic/jetpack-config": "1.4.3",
       "automattic/jetpack-autoloader": "2.8.0",
-      "myclabs/php-enum": "1.7.7"
+      "myclabs/php-enum": "1.7.7",
+      "woocommerce/action-scheduler": "3.1.6"
```
Comment (DanReyLop) [inline composer.json:20]: Let's add this dependency on `require-dev` instead so it doesn't end up in the plugin ZIP.

**Comment by dmallory42** [comment]
Closing this branch as this code was merged as part of https://github.com/Automattic/woocommerce-payments/pull/1207.

**Pull request closed without merging**